src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/service/ActivitiesService.kt

```diff
@@ -24,34 +29,52 @@ class ActivitiesService(
   private val agencyInternalLocationRepository: AgencyInternalLocationRepository,
   private val programServiceRepository: ProgramServiceRepository,
   private val availablePrisonIepLevelRepository: AvailablePrisonIepLevelRepository,
+  private val offenderBookingRepository: OffenderBookingRepository,
+  private val offenderProgramProfileRepository: OffenderProgramProfileRepository,
   private val telemetryClient: TelemetryClient,
 ) {
   companion object {
     private val log = LoggerFactory.getLogger(this::class.java)
   }
 
-  fun createActivity(dto: CreateActivityRequest): CreateActivityResponse {
+  fun createActivity(dto: CreateActivityRequest): CreateActivityResponse =
 
-    val courseActivity = activityRepository.save(mapModel(dto))
+    activityRepository.save(mapActivityModel(dto)).run {
 
-    val payRates = mapRates(dto, courseActivity)
+      payRates = mapRates(dto, this)
```

Comment:
is there a reason you mutate the course activity, could you not just supply the pay rates in the `mapModel` above?

Reply:
In fact if any pay rate has an invalid IEP level we'll create the activity but none of the pay rates then return a 400. But the activity record was created!

Reply:
payRates contains a ref back to the created CourseActivity